auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealmBuilder.java

```diff
@@ -154,6 +193,34 @@ public FileSystemSecurityRealm build() {
         if (hashCharset == null) {
             hashCharset = StandardCharsets.UTF_8;
         }
-        return new FileSystemSecurityRealm(root, nameRewriter, levels, encoded, hashEncoding, hashCharset, providers, secretKey);
+        if (providers == null) {
+            providers = INSTALLED_PROVIDERS;
+        }
+
+        if (privateKey == null ^ publicKey == null) {
+            throw ElytronMessages.log.invalidKeyPairArgument(root.toString());
+        }
+
+        // Validate KeyPair
+        if (privateKey != null) {
+            try {
+                byte[] challenge = new byte[10000];
+                ThreadLocalRandom.current().nextBytes(challenge);
+                Signature sig = Signature.getInstance("SHA256withRSA");
+                sig.initSign(privateKey);
+                sig.update(challenge);
+                byte[] signature = sig.sign();
+                sig.initVerify(publicKey);
+                sig.update(challenge);
+                boolean keyPairMatches = sig.verify(signature);
+                if (!keyPairMatches) {
+                    throw ElytronMessages.log.invalidKeyPairArgument(root.toString());
+                }
+            } catch (NoSuchAlgorithmException | InvalidKeyException | SignatureException e) {
+                throw ElytronMessages.log.invalidKeyPairArgument(root.toString());
```

Comment:
This is a bad log message. I get an error in case it should work but without any clue what's wrong:

```
[standalone@localhost:9990 /] /subsystem=elytron/key-store=applicationKS:generate-key-pair(alias=a1,distinguished-name="CN=a1")
{"outcome" => "success"}

[standalone@localhost:9990 /] /subsystem=elytron/key-store=applicationKS:store()
{
    "outcome" => "success",
    "result" => undefined
}

[standalone@localhost:9990 /] /subsystem=elytron/filesystem-realm=fs1:add(path=fs1,key-store=applicationKS,key-store-alias=a1)
{
    "outcome" => "failed",
    "failure-description" => {"WFLYCTL0080: Failed services" => {"org.wildfly.security.modifiable-security-realm.fs1" => "Failed to start service
    Caused by: java.lang.IllegalArgumentException: ELY13010: Both PrivateKey and PublicKey must be defined for realm at: fs1"}},
    "rolled-back" => true
}
```

```
11:44:17,083 ERROR [org.jboss.msc.service.fail] (MSC service thread 1-6) MSC000001: Failed to start service org.wildfly.security.modifiable-security-realm.fs1: org.jboss.msc.service.StartException in service org.wildfly.security.modifiable-security-realm.fs1: Failed to start service
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.execute(ServiceControllerImpl.java:1731)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceControllerImpl$ControllerTask.run(ServiceControllerImpl.java:1559)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1990)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1377)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.IllegalArgumentException: ELY13010: Both PrivateKey and PublicKey must be defined for realm at: fs1
	at org.wildfly.security.elytron-base@1.19.1.CR1-SNAPSHOT//org.wildfly.security.auth.realm.FileSystemSecurityRealmBuilder.build(FileSystemSecurityRealmBuilder.java:220)
	at org.wildfly.extension.elytron@19.0.0.Beta7-SNAPSHOT//org.wildfly.extension.elytron.FileSystemRealmDefinition$RealmAddHandler$1.get(FileSystemRealmDefinition.java:348)
	at org.wildfly.extension.elytron@19.0.0.Beta7-SNAPSHOT//org.wildfly.extension.elytron.FileSystemRealmDefinition$RealmAddHandler$1.get(FileSystemRealmDefinition.java:299)
	at org.wildfly.extension.elytron@19.0.0.Beta7-SNAPSHOT//org.wildfly.extension.elytron.TrivialService.start(TrivialService.java:61)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.startService(ServiceControllerImpl.java:1739)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.execute(ServiceControllerImpl.java:1701)
	... 6 more

11:44:17,084 ERROR [org.jboss.as.controller.management-operation] (management-handler-thread - 2) WFLYCTL0013: Operation ("add") failed - address: ([
    ("subsystem" => "elytron"),
    ("filesystem-realm" => "fs1")
]) - failure description: {"WFLYCTL0080: Failed services" => {"org.wildfly.security.modifiable-security-realm.fs1" => "Failed to start service
    Caused by: java.lang.IllegalArgumentException: ELY13010: Both PrivateKey and PublicKey must be defined for realm at: fs1"}}
11:44:17,085 ERROR [org.jboss.as.controller.management-operation] (management-handler-thread - 2) WFLYCTL0013: Operation ("add") failed - address: ([
    ("subsystem" => "elytron"),
    ("filesystem-realm" => "fs1")
]) - failure description: {"WFLYCTL0080: Failed services" => {"org.wildfly.security.modifiable-security-realm.fs1" => "Failed to start service
    Caused by: java.lang.IllegalArgumentException: ELY13010: Both PrivateKey and PublicKey must be defined for realm at: fs1"}}
```